llms/mlx_lm/models/cache.py

```diff
@@ -126,6 +129,88 @@ def is_trimmable(self):
         return False
 
 
+class QuantizedKVCache(_BaseCache):
+    def __init__(self, group_size: int = 64, bits: int = 8):
+        self.keys = None
+        self.values = None
+        self.offset = 0
+        self.step = 256
+        self.group_size = group_size
+        self.bits = bits
+
+    def update_and_fetch(self, keys, values):
+        B, n_kv_heads, num_steps, k_head_dim = keys.shape
+        prev = self.offset
+
+        if self.keys is None or (prev + num_steps) > self.keys[0].shape[-2]:
+            el_per_int = 8 * mx.uint32.size // self.bits
+            new_steps = (self.step + num_steps - 1) // self.step * self.step
+            shape = (B, n_kv_heads, new_steps, k_head_dim // el_per_int)
```

Comment:
One of the models (deep seek) has a different `v_head_dim` than `k_head_dim` (which is why we use two different shapes in the KV cache.  Can you do the same here so it works with that model as well?

Reply:
Good catch! Will do